rest/src/main/java/io/syndesis/rest/v1/util/Urls.java

```diff
@@ -49,6 +49,17 @@ public static URI absoluteTo(final HttpServletRequest httpRequest, final URI url
         }
     }
 
+    public static URI appHome(final HttpServletRequest httpRequest) {
+        final URI current = currentUri(httpRequest);
+
+        try {
+            return new URI(current.getScheme(), null, current.getHost(), current.getPort(), null, null, null);
```

Comment:
That's a bit dangerous when the UI is not mapped to the root context. But I also don't see a better solution.

I wonder though whether the 'current' URI also depends on the SSL termination of the route. When the route terminates at the router, then it acts as a reverse proxy to the backend, which means you will probably see the internal URI address here (and hence a redirect won't work).  

Maybe we should put the home URL also in to the application.properties and let it fill in from the openshift tempalte ?

Reply:
I guess that would be much better, note that in #546 we've configured Undertow to to acknowledge `X-Forwarded-` headers set by nginx. And in OAuth (credentials) there are a number of places absolute URIs are computed.